src/Ucommerce.Sitecore/SitecoreDataProvider/Impl/TemplateBuilders/ProductTemplatesBuilder.cs

```diff
@@ -200,15 +200,14 @@ private void UpdateSelectedCategories(FieldChange fieldChange, Product product)
 			var selectedSitecoreCategoryIds = fieldChange.Value.Split('|');
 			var selectedCategories = new List<Category>();
 
-			// Find categories from the sitecore ids
+			// Find categories from the Sitecore item GUIDs.
 			foreach (var selectedSitecoreCategoryId in selectedSitecoreCategoryIds)
 			{
-				var categoryIdKeys = IDTable.GetKeys(SitecoreConstants.SitecoreIdTablePrefix, ID.Parse(selectedSitecoreCategoryId));
-
-				foreach (var categoryIdKey in categoryIdKeys)
+				var category = ObjectFactory.Instance.Resolve<IRepository<Category>>()
+					.SingleOrDefault(x => x.Guid == Guid.Parse(selectedSitecoreCategoryId));
+				if (category != null)
 				{
-					var categoryId = Convert.ToInt32(categoryIdKey.Key.Split(';').Last());
```

Comment:
As far as I understand, the Ids stored in the IDTable had a special format, something along the lines of "something;categoryId". Since now we simply used the GUIDs, that should no longer be necessary.